local-cli/runAndroid/runAndroid.js

```diff
@@ -271,6 +269,107 @@ function startServerInNewWindow() {
   }
 }
 
+function findPreviousTerm(content, endPos) {
+  while (content[endPos] === ' ') {
+    --endPos;
+  }
+  const regex = new RegExp('\\w');
+  const word = [];
+  while (regex.exec(content[endPos])) {
+    word.push(content[endPos]);
+    --endPos;
+  }
+  return word.reverse().join('');
+}
+
+function findBuildTypes(filePath) {
+  // Read the gradle file and get list of buildTypes defined for the project.
+  const content = fs.readFileSync(filePath, 'utf8');
+  const regex = new RegExp('buildType\\s+{', 'ig');
+  const buildTypes = ['debug', 'release'];
+  const match = regex.exec(content);
+  if (!match) {
+    return buildTypes;
+  }
+  const buildTypeStartPos = regex.lastIndex;
+  let counter = 1;
+  let pos = buildTypeStartPos + 1;
+  while (counter > 0) {
+    if (content[pos] === '{') {
+      counter += 1;
+      if (counter === 2) {
+        const previousTerm = findPreviousTerm(content, pos - 1);
+        if (buildTypes.indexOf(previousTerm) === -1) {
+          buildTypes.push(previousTerm);
+        }
+      }
+    } else if (content[pos] === '}') {
+      --counter;
+    }
+    ++pos;
+  }
+  return buildTypes;
+}
+
+function splitVariant(gradleFilePath, variant) {
+  // Split the variant into buildType and flavor
+  const buildTypes = findBuildTypes(gradleFilePath, 'buildTypes', ['debug', 'release']);
+  const regexp = new RegExp(buildTypes.join('|'), 'gi');
+  const match = regexp.exec(variant);
+  let flavor = null;
+  let buildType = variant;
+  if (match) {
+    flavor = variant.substring(0, match.index);
+    buildType = variant.substring(match.index);
+  }
+  return { buildType, flavor };
+}
+
+function isSeparateBuildEnabled(gradleFilePath) {
+  // Check if separate build enabled for different processors
+  const content = fs.readFileSync(gradleFilePath, 'utf8');
+  const separateBuild = content.match(/enableSeparateBuildPerCPUArchitecture\s+=\s+([\w]+)/)[1];
```

Comment:
I think there should be a null check for `separateBuild` before you access `[1]` of it. Reason: I was integrating React Native in an existing project and by default it did not have this key in the `build.gradle` file, nor is it necessary. So it would keep throwing the wrong error because the code for getManifestFile is wrapped in a try catch. So the error that an end user actually gets is `adb invocation failed. Do you have adb in your PATH?`.